mozdownload/scraper.py

```diff
@@ -816,6 +816,28 @@ def detect_platform(self):
 
         return platform
 
+    def is_build_dir(self, dir):
+        """Return whether or not the given dir contains a build."""
+
+        url = urljoin(self.base_url, self.build_list_regex, dir)
+
+        if self.application in MULTI_LOCALE_APPLICATIONS \
+                and self.locale != 'multi':
+            url = urljoin(url, self.locale)
+
+        parser = DirectoryParser(url, authentication=self.authentication,
+                                 timeout=self.timeout_network)
+
+        pattern = re.compile(self.binary_regex, re.IGNORECASE)
+        for entry in parser.entries:
+            try:
+                pattern.match(entry).group()
+                return True
+            except:
+                # No match, continue with next entry
+                continue
+        return False
```

Comment:
@whimboo 
I now remember why I could not do it.

Please see https://github.com/mozilla/mozdownload/blob/master/mozdownload/scraper.py#L465 

This call to `parser.filter` prevents us from moving it up to base scraper. Maybe there is a way to circumvent this, but it may not look very clean.

As I mentioned in IRC, I once or twice used the built-in functools.partial method to cirumvent the constraints of only one argument. Don't know if it can work here, too.